CDCgov/prime-reportstream
# 8426 Created simple ADT A01 and OML O21 FHIR to HL7 v2 schema

This PR adds the capability to convert ADT A01 and OML O21 FHIR to HL7 v2. Note that missing segments will be converted at a later time. This is just to allow for initial conversion and there will need to be an effort to convert other segments we need.

Test Steps:
1. Verify that the integration tests pass (just verify the build passes in Github)
2. Convert a FHIR bundle to ADT A01. You first need to get a sample bundle, then change `MessageHeader.activity.coding.code` to `A01`. Run the following command from the `prime-router` folder to convert to HL7 v2:
`./gradlew --args="fhirdata -i <your sample bundle> --output-format HL7 -s metadata/hl7_mapping/ADT_A01/ADT_A01-base.yml"`
3. Convert a FHIR bundle to OML O21. You first need to get a sample bundle, then change `Bundle.entry.resource.ofType(Provenance).activity.coding.code` to `O21`. Run the following command from the `prime-router` folder to convert to HL7 v2:
`./gradlew --args="fhirdata -i <your sample bundle> --output-format HL7 -s metadata/hl7_mapping/OML_O21/OML_O21-base.yml"`

Note: You could use [this sample bundle](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.fhir) for your test

## Changes
- Allow for constants to have no string
- Added new ADT A01 schema
- Added new OML O21 schema
- Refactored some of the ORU R01 base segments to be reused
- Allow schemas to specify any HAPI HL7 v2 message type via class name (instant support for any message type)
- Enhanced some unit tests

## Checklist

### Testing

### Process

## Linked Issues
- #8426

## To Be Done
- #8445 

## Specific Security-related subjects a reviewer should pay specific attention to
